include/proteus/Config.hpp

```diff
@@ -17,6 +41,31 @@ inline int getEnvOrDefaultInt(const char *VarName, int Default) {
   return EnvValue ? std::stoi(EnvValue) : Default;
 }
 
+inline CodegenOption getEnvOrDefaultCG(const char *VarName,
+                                       CodegenOption Default) {
+
+  const char *EnvValue = std::getenv(VarName);
+  if (!EnvValue)
+    return Default;
+
+  std::string EnvValueStr{EnvValue};
+  std::transform(EnvValueStr.begin(), EnvValueStr.end(), EnvValueStr.begin(),
+                 ::tolower);
+  if (EnvValueStr == "rtc")
+    return CodegenOption::RTC;
+  if (EnvValueStr == "serial")
+    return CodegenOption::Serial;
+  if (EnvValueStr == "parallel")
+    return CodegenOption::Parallel;
+  if (EnvValueStr == "thinlto")
+    return CodegenOption::ParallelThinLTO;
+
+  Logger::outs("proteus") << "Unknown codegen option " << EnvValueStr
```

Comment:
Might want this to go to stderr rather than a log file?

Reply:
It prints to stdout but valid point. I will make an issue